### PR TITLE
Geyser battery

### PR DESCRIPTION
**Pull request by secondl1ght**
### Describe the changes you have made in this PR

I have added another battery detector for Geyser crowdfunding projects, you can now fund projects directly with Alby.

### Link this PR to an issue

No related issue but this was discussed on Twitter with @bumi 

### Type of change (Remove other not matching type)

- `feat`: New feature (non-breaking change which adds functionality)

### Screenshots of the changes (If any)

image
image

### How has this been tested?

Ran locally and was able to get Alby to detect the lightning address and other attributes on the Geyser website.

### Checklist

- [x] My code follows the style guidelines of this project and performed a self-review of my own code
- [x] New and existing tests pass locally with my changes
- [x] I checked if I need to make corresponding changes to the documentation (and made those changes if needed)

### Note

I am putting in a PR with Geyser as well the finalize this - it will work currently but I am adding more selectors so that we can use a better avatar and project title as seen in the sceenshots.

### CODE REVIEW

**Comment by secondl1ght** [comment]
The failing test seems unrelated to this PR.

**Comment by escapedcat** [comment]
> The failing test seems unrelated to this PR.

Yes, the publisher one is still flaky. Will check.

**Review by escapedcat — APPROVED** [review]
lgtm

**Review by bumi — APPROVED** [review]
this is great! 
there is one problem: geyser.fund loads the data asynchronously. This means the lightning address is not yet in the DOM when we search for the lightning address. 

that's why the icon does not change, but it works when you click on the Alby icon (then we do another check)

File: src/extension/content-script/batteries/GeyserProject.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  )?.src;
+
+  if (address) {
+    setLightningData([
+      {
+        method: "lnurl",
+        address: address,
+        ...getOriginData(),
+        description:
+          "Geyser - global bitcoin crowdfunding that helps builders and creators fund their ideas.",
```
Comment (bumi) [inline src/extension/content-script/batteries/GeyserProject.ts:32]: do we need this? it's a bit long imp. 
```suggestion
        description: "geyser.fund",
```

**Review by bumi — APPROVED** [review]

**Comment by secondl1ght** [comment]
> this is great! there is one problem: geyser.fund loads the data asynchronously. This means the lightning address is not yet in the DOM when we search for the lightning address.
> 
> that's why the icon does not change, but it works when you click on the Alby icon (then we do another check)

Ah ok, I know Geyser is looking at pre-rendering the pages eventually because this causes other issues as well like poor SEO and open graph social sharing (so this will be solved at some point).

Also Geyser will add dynamic Alby `meta` tags to the projects too, so at that point we can depreciate this battery because Alby will be able to use the `meta` tags instead which will be better for maintainability in the future. But this solution works for now!

**Comment by secondl1ght** [comment]
@bumi the Geyser PR to accommodate this has been merged - so the selectors for this PR are now live and ready to go!

**Pull request merged**